src/main/java/org/broadinstitute/consent/http/cloudstore/GCSService.java

```diff
@@ -138,6 +143,23 @@ public Map<BlobId, InputStream> getDocuments(List<BlobId> blobIds) throws NotFou
     }
   }
 
+  /**
+   * Read JSON file from GCS bucket
+   *
+   * @param blobIdName String value of the document blob id name
+   * @param valueType  Class type of the object to be read
+   * @return Object of the specified type
+   */
+  public <T> T readJsonFileFromBucket(String blobIdName, Class<T> valueType) throws IOException {
+    try (InputStream is = getDocument(blobIdName)) {
```

Comment:
It's not part of this PR, but looking at the rest of this class, it's not using `Optional` correctly. It's building an `Optional` using `Optional.of()`, not `Optional.ofNullable()`. If you use `.of()`, the contents can't be `null` and are always present.

I don't think the GCS APIs will return `null` because if a blob isn't found an exception is thrown. If that's true, `Optional` can be removed from this class.